src/button-group/button-group.js

```diff
@@ -97,7 +96,7 @@ export class ButtonGroupRoot extends React.Component<{|
                   this.childRefs[nextIndex].current.focus();
               }
             },
-            kind,
+            kind: isSelected(selected, index) ? KIND.primary : kind,
```

Comment:
What do you want to happen if the dev specifies a `kind` that's not `primary`? `tertiary` for example. Should that also have primary buttons when it's selected?